notifiers/providers/dingtalk.py

```diff
@@ -0,0 +1,91 @@
+from ..core import Response
+from ..utils import requests
+from ..utils.helpers import snake_to_camel_case
+
+
+class DingTalk(Provider):
+    """Send DingTalk notifications"""
+
+    base_url = "https://oapi.dingtalk.com/robot/send?access_token={}"
+    site_url = "https://oapi.dingtalk.com/"
+    path_to_errors = ("message",)
+    name = "dingtalk"
+
+    _required = {"required": ["access_token", "message"]}
+    _schema = {
+        "type": "object",
+        "properties": {
+            "access_token": {
+                "type": "string",
+                "title": "access token to pair a channel to receive notification",
+            },
+            "msg_type": {
+                "type": "string",
+                "title": "choose a message type, these type are supported: text, markdown, link, action_card",
+                "enum": ["text", "markdown", "link", "action_card"],
+            },
+            "message": {
+                "type": "string",
+                "title": "This is the text that will be posted to the dingtalk",
+                "maxLength": 4096,
+            },
+            "msg_title": {
+                "type": "string",
+                "title": "title for markdown message and card message",
+            },
+            "msg_url": {
+                "type": "string",
+                "format": "uri",
+                "title": "url for markdown message",
+            },
+            "msg_btn_title": {
+                "type": "string",
+                "title": "title for card message button, like 'Read more.'",
+            },
+            "msg_btn_url": {
+                "type": "string",
+                "format": "uri",
+                "title": "url for card message button",
+            },
+        },
+        "additionalProperties": False,
+    }
+
+    @property
+    def defaults(self) -> dict:
+        return {"msg_type": "text"}
+
+    def _prepare_data(self, data: dict) -> dict:
+        text = data.pop("message")
+        mapping_key = {
+            "msg_title": "title",
+            "msg_url": "messageUrl",
+            "msg_btn_title": "singleTitle",
+            "msg_btn_url": "singleURL",
+        }
+
+        new_data = {
+            "access_token": data.pop("access_token"),
+            "msgtype": data.pop("msg_type"),
+        }
+
+        if new_data["msgtype"] != "text":
+            camel_case_key = snake_to_camel_case(new_data["msgtype"])
+            new_data["msgtype"] = camel_case_key[0].lower() + camel_case_key[1:]
+
+            new_data[new_data["msgtype"]] = {"text": text}
+            for key in data:
+                new_data[new_data["msgtype"]][mapping_key[key]] = data[key]
+        else:
+            new_data["text"] = {"content": text}
+
+        return new_data
+
+    def _send_notification(self, data: dict) -> Response:
+        access_token = data.pop("access_token")
+        url = self.base_url.format(access_token)
+        response, errors = requests.post(
```

Comment:
please remove the `?access_token=` from the url and pass it as `params={'access_token': access_token)` to the request. Using `params` correectly encodes URL params if needed

Reply:
@liiight Hi Or Carmi, i sent a access token to your email`python.notifiers@gmail.com`, please to check.